app/src/main/java/com/github/swent/echo/viewmodels/HomeScreenViewModel.kt

```diff
@@ -125,13 +125,19 @@ constructor(
 
     fun onFollowedTagClicked(tag: Tag) {
         if (_followedTags.value.contains(tag)) {
-            if (_selectedTagId.value == tag.tagId) {
-                _selectedTagId.value = null
-                followedTagFilter = getTagsAndSubTags(_followedTags.value).toList()
+            if (_selectedTagIds.value.contains(tag.tagId)) {
+                _selectedTagIds.value = _selectedTagIds.value.minus(tag.tagId)
             } else {
-                _selectedTagId.value = tag.tagId
-                followedTagFilter = getTagsAndSubTags(listOf(tag)).toList()
+                _selectedTagIds.value = _selectedTagIds.value.plus(tag.tagId)
```

Comment:
hint: you could use `-` and `+` here instead of the explicit `.minus()` and `.plus()` methods :)